src/weather.cpp

```diff
@@ -799,13 +801,62 @@ nc_color get_wind_color( double windpower )
     return windcolor;
 }
 
+std::string get_shortdirstring( int angle )
+{
+    std::string dirstring;
+    int dirangle = angle;
+    if( dirangle <= 23 || dirangle > 338 ) {
+        dirstring = ( "N" );
```

Comment:
Direction handling code looks like reinventing the wheel in a bad way. It would be nice to fix it if you have time. Please take a look at `enum direction` in line.h and the related stuff. This would help to eliminate these two functions completely.

Reply:
Would the best approach perhaps be to generate wind direction as a vector from the point of generation, then run that through the line.cpp stuff?